ui/shared/AccountSelector.qml

```diff
@@ -0,0 +1,201 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtGraphicalEffects 1.13
+import "../imports"
+
+Item {
+    id: root
+    property string label: qsTr("Choose account")
+    property var accounts
+    property var selectedAccount: {
+        "address": "", "name": "", "iconColor": "", "fiatBalance": ""
+    }
+    height: select.height + selectedAccountDetails.height
+    // set to asset symbol to display asset's balance top right
+    // NOTE: if this asset is not selected as a wallet token in the UI, then
+    // nothing will be displayed
+    property string showAssetBalance: ""
+
+    Repeater {
+        visible: showAssetBalance !== ""
+        model: selectedAccount.assets
+        delegate: StyledText {
+            visible: symbol === root.showAssetBalance.toUpperCase()
+            anchors.bottom: select.top
+            anchors.bottomMargin: -18 
+            anchors.right: parent.right
+            text: "Balance: " + (parseFloat(value) === 0.0 ? "0" : value) + " " + symbol
+            color: parseFloat(value) === 0.0 ? Style.current.red : Style.current.darkGrey
+            font.pixelSize: 13
+            height: 18
+        }
+    }
+    Select {
+        id: select
+        icon: "../app/img/walletIcon.svg"
+        iconColor: selectedAccount.iconColor || Style.current.blue
+        label: root.label
+        selectedText: selectedAccount.name
+        model: root.accounts
+
+        menu.delegate: menuItem
+        menu.onOpened: {
+            selectedAccountDetails.visible = false
+        }
+        menu.onClosed: {
+            selectedAccountDetails.visible = true
+        }
+    }
+
+    Row {
+        id: selectedAccountDetails
+        anchors.top: select.bottom
+        anchors.topMargin: 8
+        anchors.left: parent.left
+        anchors.leftMargin: 2
+
+        StyledText {
+            id: textSelectedAddress
+            text: selectedAccount.address
+            font.pixelSize: 12
+            elide: Text.ElideMiddle
+            height: 16
+            width: 80
+            color: Style.current.darkGrey
+        }
+        StyledText {
+            text: " • "
+            font.pixelSize: 12
+            height: 16
+            color: Style.current.darkGrey
+        }
+        StyledText {
+            text: selectedAccount.fiatBalance + " " + walletModel.defaultCurrency.toUpperCase()
+            font.pixelSize: 12
+            height: 16
+            color: Style.current.darkGrey
+        }
+    }
+
+    Component {
+        id: menuItem
+        MenuItem {
+            id: itemContainer
+            property color bgColor: Style.current.white
```

Comment:
This one doesn't work in dark mode

Reply:
Just confirming what Pascal said:
```suggestion
            property color bgColor: Style.current.background
```

Reply:
Ah, if it doesn't work in dark mode, likely is because I created the dark mode PR at the same time which fixes all dark mode issues.

I don't agree about the background though. I think we should be declaring these as they are specified in "light" mode, such as white. White gets automatically swapped when dark mode is activated, as per the design. 

Please see https://github.com/status-im/nim-status-client/pull/698 for more info.

Reply:
Agreed, will handle this in updates to https://github.com/status-im/nim-status-client/pull/698.

Reply:
Is it ok to handle this in https://github.com/status-im/nim-status-client/pull/698?

Reply:
Changed as per @jrainville's suggestion. However, a review of the dark mode operation will be done in #698.